src/Gui/DlgSettingsWorkbenchesImp.h

```diff
@@ -31,42 +31,71 @@ class QCheckBox;
 
 namespace Gui {
 namespace Dialog {
-class Ui_DlgSettingsLazyLoaded;
-
+class Ui_DlgSettingsWorkbenches;
 
 /**
- * The DlgSettingsLazyLoadedImp class implements a pseudo-preference page explain why
+ * The DlgSettingsWorkbenchesImp class implements a pseudo-preference page explain why
  * the remaining preference pages aren't loaded yet, and to help the user do so on demand.
  * \author Jürgen Riegel
  */
-class DlgSettingsLazyLoadedImp : public PreferencePage
+class DlgSettingsWorkbenchesImp : public PreferencePage
 {
     Q_OBJECT
 
 public:
-    explicit DlgSettingsLazyLoadedImp( QWidget* parent = nullptr );
-    ~DlgSettingsLazyLoadedImp() override;
+    explicit DlgSettingsWorkbenchesImp( QWidget* parent = nullptr );
+    ~DlgSettingsWorkbenchesImp() override;
 
     void saveSettings() override;
     void loadSettings() override;
 
-protected Q_SLOTS:
-    void onLoadClicked(const QString& wbName);
+    static QStringList getEnabledWorkbenches();
+    static QStringList getDisabledWorkbenches();
+
+    void setStartWorkbenchComboItems();
+
+    std::vector<std::string> _backgroundAutoloadedModules;
+    std::string _startupModule;
 
 protected:
-    void buildUnloadedWorkbenchList();
+    void buildWorkbenchList();
     void changeEvent(QEvent *e) override;
 
 private:
-    std::unique_ptr<Ui_DlgSettingsLazyLoaded> ui;
-    static const uint WorkbenchNameRole;
+    void addWorkbench(const QString& it, bool enabled);
 
-    std::vector<std::string> _backgroundAutoloadedModules;
-    std::string _startupModule;
-    std::map<QString, QCheckBox*> _autoloadCheckBoxes;
+    void saveWorkbenchSelector();
+    void loadWorkbenchSelector();
+
+    void onStartWbChangedClicked(int index);
+
+    std::unique_ptr<Ui_DlgSettingsWorkbenches> ui;
+};
+
+class wbListItem : public QWidget
```

Comment:
I guess according usage scope of this class, it could have been directly defined in CPP file.

Reply:
Better use it. You asked to me a question in one of my PR just few days ago 😉

Reply:
https://github.com/FreeCAD/FreeCAD/pull/8952#discussion_r1141282500

Reply:
Oh right. Sorry I forgot about this point!